auth/env_keychain.go

```diff
@@ -156,28 +209,11 @@ func authConfigToHeader(config *authn.AuthConfig) (string, error) {
 		return fmt.Sprintf("Basic %s", encoded), nil
 	}
 
-	return "", nil
-}
-
-var (
-	basicAuthRegExp  = regexp.MustCompile("(?i)^basic (.*)$")
-	bearerAuthRegExp = regexp.MustCompile("(?i)^bearer (.*)$")
-)
-
-func authHeaderToConfig(header string) (*authn.AuthConfig, error) {
-	if matches := basicAuthRegExp.FindAllStringSubmatch(header, -1); len(matches) != 0 {
-		return &authn.AuthConfig{
-			Auth: matches[0][1],
-		}, nil
-	}
-
-	if matches := bearerAuthRegExp.FindAllStringSubmatch(header, -1); len(matches) != 0 {
-		return &authn.AuthConfig{
-			RegistryToken: matches[0][1],
-		}, nil
+	if config.IdentityToken != "" {
+		return fmt.Sprintf("X-Identity %s", config.IdentityToken), nil
```

Comment:
This is a bit icky. If we are given an identity token we need some place to put it, but there is no proper header for it.

Reply:
> This is a bit icky. If we are given an identity token we need some place to put it, but there is no proper header for it.

I had that question reading through, but quickly realized why this hack is necessary given the current interface for passing registry auth details from pack to the lifecycle. I think a comment explaining the weirdness might be appropriate to explain what is happening to the uninitiated and to our future selves.